optimum/habana/trl/trainer/ppo_trainer.py

```diff
@@ -0,0 +1,881 @@
+# Copyright 2022 The HuggingFace Team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import math
+import time
+import typing
+import warnings
+from contextlib import nullcontext
+from typing import Callable, List, Optional, Union
+
+import numpy as np
+import torch
+from accelerate.utils import ProjectConfiguration
+from datasets import Dataset
+from torch.optim import Adam
+from transformers import (
+    DataCollatorForLanguageModeling,
+    PreTrainedTokenizer,
+    PreTrainedTokenizerBase,
+    PreTrainedTokenizerFast,
+)
+from trl import PPOTrainer
+from trl.core import (
+    WANDB_PADDING,
+    PPODecorators,
+    convert_to_scalar,
+    logprobs_from_logits,
+    stack_dicts,
+    stats_to_np,
+)
+from trl.import_utils import is_npu_available, is_torch_greater_2_0, is_xpu_available
+from trl.models import SUPPORTED_ARCHITECTURES, PreTrainedModelWrapper, create_reference_model
+from trl.trainer import AdaptiveKLController, BaseTrainer, FixedKLController, RunningMoments
+
+from optimum.habana.utils import set_seed
+
+from . import GaudiPPOConfig
+
+
+class GaudiPPOTrainer(PPOTrainer, BaseTrainer):
```

Comment:
Why inheriting from both `PPOTrainer` and `BaseTrainer`? `PPOTrainer` already inherits from `BaseTrainer`: https://github.com/huggingface/trl/blob/09ca7607d50714a80c9f0feb206b29f1c5a49bab/trl/trainer/ppo_trainer.py#L109